packages/sit-onyx/src/demos/FormDemo/FormDemo.vue

```diff
@@ -0,0 +1,79 @@
+<script lang="ts" setup>
+import { TestInput } from "@/index";
+
+/**
+https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation
+- required: Specifies whether a form field needs to be filled in before the form can be submitted.
+- minlength and maxlength: Specifies the minimum and maximum length of textual data (strings).
+- min and max: Specifies the minimum and maximum values of numerical input types.
+- type: Specifies
+whether the data needs to be a number, an email address, or some other specific preset type.
+-pattern: Specifies a regular expression that defines a pattern the entered data needs to follow.
+
+https://developer.mozilla.org/en-US/docs/Web/CSS/:valid
+- The element matches the :valid or :invalid CSS pseudo-class
+
+Build-in errors like:
+https://developer.mozilla.org/en-US/docs/Web/API/ValidityState/badInput
+https://developer.mozilla.org/en-US/docs/Web/API/ValidityState/valueMissing
+https://developer.mozilla.org/en-US/docs/Web/API/ValidityState/patternMismatch
+- const input = document.getElementById("age");
+- => input.validity.badInput
+*/
+
+const props = defineProps<{
+  formData: {
+    defaultInput: string;
+    requiredInput: string;
+    minlengthInput: string;
+    maxInput: string;
+    typeInput: string;
+    patternInput: string;
+  };
+}>();
+
+const emit = defineEmits<{
+  submit: [data: object];
+}>();
+
+const { formData } = props;
+
+const onSubmit = (event: Event) => {
+  event.preventDefault();
+  emit("submit", formData);
+};
+</script>
+
+<template>
+  <form @submit="onSubmit($event)">
+    <TestInput v-model="formData.defaultInput" label="Default"></TestInput>
```

Comment:
Don't we have a eslint rule for auto fixing self closing tags?
```suggestion
    <TestInput v-model="formData.defaultInput" label="Default" />
```

Reply:
I thought so too, but pnpm lint-fix does not change it somehow.

Reply:
We need to add this to our root `.eslintrc`:
```js
rules: {
  "vue/html-self-closing": [
    // see https://eslint.vuejs.org/rules/html-self-closing
    "error",
    {
      html: {
        void: "always",
        normal: "never",
        component: "always",
      },
    },
  ],
},

```